File: ruby/RSAExample.rb

require 'openssl'
require 'base64'
require 'json'

rawData = {
  :partnerCode => 'MOMO5G4K20180302',
  :requestId => '410458',
  :amount => 110000,
  :signature => '410458'
}

# load key
key = OpenSSL::PKey::RSA.new File.read 'mykey.pem'
# encryptdata
encrypted_string = key.public_encrypt(JSON.pretty_generate(rawData))
encoded_str = Base64.encode64(encrypted_string)

File: python/RSAExample.py
from Crypto.PublicKey import RSA
from Crypto.Cipher import PKCS1_v1_5 as Cipher_PKCS1_v1_5
from base64 import b64encode
import json

# load key
f = open('mykey.pem', 'r')
key = RSA.importKey(f.read())

# encrypt data
rowData = {
  'partnerCode': 'MOMO5G4K20180302',
  'partnerRefId': 'c1f10470-8a68-11e8-9a57',
  'amount': 101000,
  'partnerTransId': 'c1f10470-8a68-11e8-9a57'
}
cipher = Cipher_PKCS1_v1_5.new(key)
cipher_text = cipher.encrypt(json.dumps(rowData).encode())
emsg = b64encode(cipher_text)
print "Encrypted: " + emsg